ReactWindows/ReactNative/Storage/FutureAccessList.cs

```diff
@@ -0,0 +1,146 @@
+            return Scheme + Uri.EscapeUriString(token);
+        }
+
+        /// <summary>
+        /// Removes the file or folder associated with this URI.
+        /// </summary>
+        public static void Remove(string uri)
+        {
+            if (TryExtractAccessToken(uri, out var token))
+            {
+                StorageApplicationPermissions.FutureAccessList.Remove(token);
+            }
+        }
+
+        /// <summary>
+        /// Resolves a URI returned by the Add(file) method to the file object.
+        /// The file object remains in the cache and can be retrieved again later.
+        /// </summary>
+        public static async Task<StorageFile> GetFileAsync(string uri)
+        {
+            var token = ExtractAccessToken(uri);
+            return await StorageApplicationPermissions.FutureAccessList.GetFileAsync(token);
+        }
+
+        /// <summary>
+        /// Resolves a URI returned by the Add(folder) method to the folder object.
+        /// The folder object remains in the cache and can be retrieved again later.
+        /// </summary>
+        public static async Task<StorageFolder> GetFolderAsync(string uri)
+        {
+            var token = ExtractAccessToken(uri);
+            return await StorageApplicationPermissions.FutureAccessList.GetFolderAsync(token);
+        }
+
+        /// <summary>
+        /// Checks if FAL contains a file or folder with the given URI.
+        /// Returns false If the URI is invalid.
+        /// </summary>
+        public static bool ContainsItem(string uri)
+        {
+            if (TryExtractAccessToken(uri, out var token))
+            {
+                return StorageApplicationPermissions.FutureAccessList.ContainsItem(token);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static bool TryExtractAccessToken(string uri, out string token)
+        {
+            if (!uri.StartsWith(Scheme))
+            {
+                token = null;
+                return false;
+            }
+            else
+            {
+                token = Uri.UnescapeDataString(uri.Substring(Scheme.Length));
+                return true;
+            }
+        }
+
+        private static string ExtractAccessToken(string uri)
+        {
+            if (!TryExtractAccessToken(uri, out var token))
+            {
+                throw new ArgumentException("Invalid future access list URI: " + uri);
+            }
+            else
+            {
+                return token;
+            }
+        }
+
+        private static string GetAccessToken(IStorageItem item)
+        {
+            var futureAccessList = StorageApplicationPermissions.FutureAccessList;
+            var existingEntries = futureAccessList.Entries;
+            var capacity = futureAccessList.MaximumItemsAllowed;
+            var now = DateTime.UtcNow;
+
+            var expiry = existingEntries.Count >= capacity / 2 ?
```

Comment:
>expiry [](start = 16, length = 6)

Seems like this should be configurable by the apps. Would be a good argument for a `FutureAccessListModule`. Doesn't need to be done here, but please at least add an issue to GitHub.

Reply:
I'd say this entire logic needs to be rewritten (that our discussion about LFU cache). However for now I'd rather keep it simple and optimize only we need to.

Reply:
We can use #1802 for tracking.